File: python/src/problems1_100/sort_colors.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def sort_colors(self, nums: list[int]) -> None:
        pass


class Default(Problem):
    @override
    def sort_colors(self, nums: list[int]) -> None:
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[j] >= nums[i]:
                    continue
                temp: int = nums[j]
                nums[j] = nums[i]
                nums[i] = temp


SOLUTIONS: list[Problem] = [Default()]

File: javascript/src/problems1_100/sort-colors.js
class Problem {
  sortColors(nums) {
    throw new Error(`Not implemented (${nums}).`);
  }
}

class Default extends Problem {
  sortColors(nums) {
    for (let i = 0; i < nums.length - 1; i++) {
      for (let j = i + 1; j < nums.length; j++) {
        if (nums[j] >= nums[i]) {
          continue;
        }
        const temp = nums[j];
        nums[j] = nums[i];
        nums[i] = temp;
      }
    }
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;
